app/auth.tsx

```diff
@@ -1,25 +1,38 @@
-import React from "react";
+import React, { useEffect } from "react";
 import { Button, StyleSheet } from "react-native";
 import { ThemedView } from "@/components/ThemedView";
 import { ThemedText } from "@/components/ThemedText";
 import { useSession } from "@/hooks/ctx";
 import { router } from "expo-router";
+import { initDB, insertUser } from "./(tabs)/db-service";
 
 export default function AuthScreen() {
     const { signIn } = useSession();
 
+    // Initialize database when the app starts
+    useEffect(() => {
```

Comment:
This check should be moved to the _layout.tsx to ensure that the database is initialized before accessing the app and if not it will initialize.